test/router/path_trie_crud_test.dart

```diff
@@ -300,6 +355,72 @@ void main() {
         expect(trie.lookup(pathABC), isNull);
         expect(trie.lookup(pathABD)?.value, 2);
       });
+
+      test(
+          'Given a trie with an existing wildcard path /data/* and value, '
+          'when remove is called for /data/*, '
+          'then the value is removed and lookup for matching paths returns null',
+          () {
+        final pathDefinition = NormalizedPath('/data/*');
+        trie.add(pathDefinition, 10);
+        trie.add(NormalizedPath('/data/fixed'), 20); // Sibling literal
+
+        final removedValue = trie.remove(pathDefinition);
+
+        expect(removedValue, equals(10));
+        expect(trie.lookup(NormalizedPath('/data/any')), isNull,
+            reason: 'Wildcard path should be removed.');
+        expect(trie.lookup(NormalizedPath('/data/fixed'))?.value, 20,
+            reason: 'Sibling literal path should be unaffected.');
+      });
+
+      test(
+          'Given a trie, '
+          'when remove is called for a wildcard path /data/* that does not exist as a defined route, '
+          'then null is returned and no other paths are affected', () {
+        final pathDefinition = NormalizedPath('/data/*');
+        trie.add(
+            NormalizedPath('/other/*'), 1); // Add a different wildcard path
+
+        final removedValue = trie.remove(pathDefinition);
+
+        expect(removedValue, isNull);
+        expect(trie.lookup(NormalizedPath('/other/something'))?.value, 1);
+      });
+
+      test(
+          'Given a trie with an existing tail path /files/** and value, '
+          'when remove is called for /files/**, '
+          'then the value is removed and lookup for matching paths returns null',
+          () {
+        final pathDefinition = NormalizedPath('/files/**');
+        trie.add(pathDefinition, 30);
+        trie.add(NormalizedPath('/files/specific/file.txt'),
+            40); // More specific child
+
+        final removedValue = trie.remove(pathDefinition);
+
+        expect(removedValue, equals(30));
+        expect(trie.lookup(NormalizedPath('/files/a/b/c')), isNull,
+            reason: 'Tail path should be removed.');
+        final specificLookup =
+            trie.lookup(NormalizedPath('/files/specific/file.txt'));
+        expect(specificLookup?.value, 40,
+            reason:
+                'More specific path should ideally remain if structured correctly.');
```

Comment:
```suggestion
                'More specific path should remain');
```

Or what does "structured correctly" mean? Is it not just "the code working correctly" / according to spec?